service/awsconfig/v10/resource/lifecycle/create.go

```diff
@@ -258,11 +258,11 @@ func (r *Resource) createNodeConfig(ctx context.Context, customObject providerv1
 	return nil
 }
 
-func (r *Resource) deleteNodeConfig(ctx context.Context, privateDNS string) error {
+func (r *Resource) deleteNodeConfig(ctx context.Context, nodeConfig corev1alpha1.NodeConfig) error {
 	r.logger.LogCtx(ctx, "level", "debug", "message", "deleting node config for guest cluster node")
 
-	n := v1.NamespaceDefault
-	i := privateDNS
+	n := nodeConfig.GetNamespace()
+	i := nodeConfig.GetName()
```

Comment:
When deleting we use the namespace and name provided by the object.

Reply:
We should do it always actually.